fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java

```diff
@@ -885,12 +880,10 @@ public void checkSchemaChangeAllowed(Column other) throws DdlException {
             }
         }
 
-        if ((getDataType() == PrimitiveType.VARCHAR && other.getDataType() == PrimitiveType.VARCHAR) || (
-                getDataType() == PrimitiveType.CHAR && other.getDataType() == PrimitiveType.VARCHAR) || (
-                getDataType() == PrimitiveType.CHAR && other.getDataType() == PrimitiveType.CHAR)) {
-            if (getStrLen() > other.getStrLen()) {
-                throw new DdlException("Cannot shorten string length");
-            }
+        // Nested types only support changing the order and increasing the length of the nested char type
+        // Char-type only support length growing
+        if (!type.isSupportSchemaChangeForCharType(other.type)) {
```

Comment:
what about type is Array other.type is Int, the exception error message is not correct